recipes/gtsam/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+                        "WITH_EIGEN_MKL_OPENMP": False,
+                        "THROW_CHEIRALITY_EXCEPTION": True,
+                        "ALLOW_DEPRECATED_SINCE_V4": True,
+                        "TYPEDEF_POINTS_TO_VECTORS": False,
+                        "SUPPORT_NESTED_DISSECTION": False,
+                        "TANGENT_PREINTEGRATION": False,
+                        "BUILD_WRAP": True,
+                        "WRAP_SERIALIZATION": True,
+                        "BUILD_UNSTABLE": True,
+                        "DISABLE_NEW_TIMERS": False,
+                        "BUILD_TYPE_POSTFIXES": True,
+                        "Boost_USE_STATIC_LIBS": True,
+                        "INSTALL_MATLAB_TOOLBOX": False,
+                        "INSTALL_CYTHON_TOOLBOX": False,
+                        "INSTALL_CPPUNITLITE": True,
+                        "INSTALL_GEOGRAPHICLIB": False}
+    generators = "cmake"
+    exports_sources = ["patches/*","CMakeLists.txt"]
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.verbose = False
+            self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+            self._cmake.definitions["GTSAM_USE_QUATERNIONS"] = self.options.USE_QUATERNIONS
+            self._cmake.definitions["GTSAM_POSE3_EXPMAP"] = self.options.POSE3_EXPMAP
+            self._cmake.definitions["GTSAM_ROT3_EXPMAP"] = self.options.ROT3_EXPMAP
+            self._cmake.definitions["GTSAM_ENABLE_CONSISTENCY_CHECKS"] = self.options.ENABLE_CONSISTENCY_CHECKS
+            self._cmake.definitions["GTSAM_WITH_TBB"] = self.options.WITH_TBB
+            self._cmake.definitions["GTSAM_WITH_EIGEN_MKL"] = self.options.WITH_EIGEN_MKL
+            self._cmake.definitions["GTSAM_WITH_EIGEN_MKL_OPENMP"] = self.options.WITH_EIGEN_MKL_OPENMP
+            self._cmake.definitions["GTSAM_THROW_CHEIRALITY_EXCEPTION"] = self.options.THROW_CHEIRALITY_EXCEPTION
+            self._cmake.definitions["GTSAM_ALLOW_DEPRECATED_SINCE_V4"] = self.options.ALLOW_DEPRECATED_SINCE_V4
+            self._cmake.definitions["GTSAM_TYPEDEF_POINTS_TO_VECTORS"] = self.options.TYPEDEF_POINTS_TO_VECTORS
+            self._cmake.definitions["GTSAM_SUPPORT_NESTED_DISSECTION"] = self.options.SUPPORT_NESTED_DISSECTION
+            self._cmake.definitions["GTSAM_TANGENT_PREINTEGRATION"] = self.options.TANGENT_PREINTEGRATION
+            self._cmake.definitions["GTSAM_BUILD_UNSTABLE"] = self.options.BUILD_UNSTABLE
+            self._cmake.definitions["GTSAM_DISABLE_NEW_TIMERS"] = self.options.DISABLE_NEW_TIMERS
+            self._cmake.definitions["GTSAM_BUILD_TYPE_POSTFIXES"] = self.options.BUILD_TYPE_POSTFIXES
+            self._cmake.definitions["GTSAM_BUILD_TESTS"] = False
+            self._cmake.definitions["Boost_USE_STATIC_LIBS"] = self.options.Boost_USE_STATIC_LIBS
```

Comment:
Instead of adding a `Boost_USE_STATIC_LIBS` option, use `self.options['boost'].shared`